components/layout/navbar/profile-menu/handle-login.ts

```diff
@@ -0,0 +1,18 @@
+"use client";
+
+import { addUser } from "@/server/admin-function/add-user";
+
+export const handleLogin = async (domain: string, session: any) => {
+    try {
+      if (session?.user) {
+        const userId = session.user.id as string;
+        const email = session.user.email as string;
+        const name = session.user.name as string;
+        const image = session.user.image as string;
```

Comment:
as string is not a good validation, if it doesn't exist you don't verify it and the addUser will be called, instead use default value and conditions

Reply:
default value for a user creation? google auth always returns these value, google with return defaut image if the user doesnt have a picture by exemple, do i not only need to check if the session.user exist?